### ISSUE TITLE
[Phase 4] Evaluation Framework - Detection Metrics

### ISSUE THREAD

**Issue by Sakeeb91** [Sakeeb91/gravitational-wave-detection#5]
## Objective

Build evaluation framework with detection-specific metrics and visualization.

**Part of:** #1

## Dependencies

- #4 (Model Implementation) must be complete

### Metrics Calculator
- [ ] Create `src/evaluation/metrics.py` with `DetectionMetrics`
- [ ] Implement AUC-ROC
- [ ] Add False Alarm Rate (FAR) calculation
- [ ] Compute detection efficiency at fixed FAR

### Visualization
- [ ] Create `src/evaluation/visualizer.py`
- [ ] Plot ROC curves
- [ ] Visualize signal spectrograms
- [ ] Show detection examples

### Report Generation
- [ ] Create `src/evaluation/reporter.py`
- [ ] Generate markdown summary

### Detection Metrics

```python
# src/evaluation/metrics.py
import numpy as np
from sklearn.metrics import roc_auc_score, roc_curve

class DetectionMetrics:
    def __init__(self, y_true, y_proba):
        self.y_true = np.array(y_true)
        self.y_proba = np.array(y_proba)

    def auc(self):
        """Compute AUC-ROC."""
        return roc_auc_score(self.y_true, self.y_proba)

    def efficiency_at_far(self, target_far=1e-3):
        """Detection efficiency at fixed false alarm rate."""
        fpr, tpr, thresholds = roc_curve(self.y_true, self.y_proba)

        # Find threshold for target FAR
        idx = np.argmin(np.abs(fpr - target_far))
        return tpr[idx], thresholds[idx]

    def far_at_efficiency(self, target_eff=0.9):
        """False alarm rate at fixed detection efficiency."""
        fpr, tpr, thresholds = roc_curve(self.y_true, self.y_proba)

        # Find threshold for target efficiency
        idx = np.argmin(np.abs(tpr - target_eff))
        return fpr[idx], thresholds[idx]

    def optimal_threshold(self):
        """Find threshold that maximizes TPR - FPR."""
        fpr, tpr, thresholds = roc_curve(self.y_true, self.y_proba)
        idx = np.argmax(tpr - fpr)
        return thresholds[idx]
```

### Signal Visualizer

```python
# src/evaluation/visualizer.py
import matplotlib.pyplot as plt
import numpy as np

class GWVisualizer:
    def __init__(self, fs=2048):
        self.fs = fs

    def plot_strain(self, strain, title="Strain Data"):
        """Plot time series strain from all detectors."""
        fig, axes = plt.subplots(3, 1, figsize=(12, 6), sharex=True)
        t = np.arange(strain.shape[1]) / self.fs

        detector_names = ['H1', 'L1', 'V1']
        for i, (ax, name) in enumerate(zip(axes, detector_names)):
            ax.plot(t, strain[i], lw=0.5)
            ax.set_ylabel(name)

        axes[-1].set_xlabel('Time (s)')
        axes[0].set_title(title)
        plt.tight_layout()
        return fig

    def plot_spectrogram(self, strain, detector_idx=0):
        """Plot spectrogram of strain data."""
        from scipy.signal import spectrogram

        f, t, Sxx = spectrogram(strain[detector_idx], fs=self.fs, nperseg=256)

        plt.figure(figsize=(10, 4))
        plt.pcolormesh(t, f, 10 * np.log10(Sxx + 1e-10), shading='auto')
        plt.ylabel('Frequency (Hz)')
        plt.xlabel('Time (s)')
        plt.ylim(20, 500)
        plt.colorbar(label='Power (dB)')
        return plt.gcf()
```

## Definition of Done

- [ ] AUC matches sklearn (validated)
- [ ] FAR/efficiency trade-off documented
- [ ] Signal visualizations show chirp patterns
- [ ] ROC curves with proper labels

## Technical Notes

- FAR in GW astronomy is typically quoted as events per year
- For G2Net: FAR = FPR × (time_analyzed / 2 seconds)
- Detection efficiency = TPR = recall
